common/dbconfig.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "common/config.h"
+#include "common/jsoncpp/include/json/json.h"
+
+/**
+ * {
+ *   "dataset" : {
+ *      // dataset level configs
+ *        "ad_query_join" : {
+ *              "ack_mode" : 1,
+ *       }
+ *    }
+ * }
+ */
+
+namespace common {
+
+struct DBConfig {
+  std::unordered_map<std::string, Config> dataSet;
```

Comment:
should we rename to something like `segmentToConfigMap`, I feel like `dataSet` means a dataset struct.
e.g. at dbconfig.cpp L63, the `dataset` means `dataset name`. 

Now, `dataset` could cause confusion when read the code

Reply:
good point will fix the dataset variable..